packages/ppx/test/native/At_rule_test.re

```diff
@@ -5,87 +5,196 @@ let loc = Location.none;
     [%expr
       CssJs.style([|
-        CssJs.media({js|(max-height: |js} ++ wat ++ {js|)|js}, [||]),
+        CssJs.media({js|(not (color)) and (not (hover)) |js}, [||]),
       |])
     ],
   ),
   (
-    "($(wat))",
-    [%expr [%cx "@media ($(wat)) {}"]],
+    "screen and (min-width: 33px) and (max-height: 15rem)",
     [%expr
-      CssJs.style([|CssJs.media({js|(|js} ++ wat ++ {js|)|js}, [||])|])
+      [%cx "@media screen and (min-width: 33px) and (max-height: 15rem) {}"]
+    ],
+    [%expr
+      CssJs.style([|
+        CssJs.media(
+          {js|screen and (min-width: 33px) and (max-height: 15rem) |js},
+          [||],
+        ),
+      |])
     ],
   ),
   (
-    "$(wat)",
-    [%expr [%cx "@media $(wat) {}"]],
-    [%expr CssJs.style([|CssJs.media(wat, [||])|])],
+    "all and (not(hover))",
+    [%expr [%cx "@media all and (not(hover)){}"]],
+    [%expr CssJs.style([|CssJs.media({js|all and (not(hover))|js}, [||])|])],
   ),
   (
-    "(max-height: $(wat))",
+    "not all and (monochrome)",
+    [%expr [%cx "@media not all and (monochrome) {}"]],
+    [%expr
+      CssJs.style([|CssJs.media({js|not all and (monochrome) |js}, [||])|])
+    ],
+  ),
+  (
+    "screen, print and (color)",
+    [%expr [%cx "@media screen, print and (color) {}"]],
+    [%expr
+      CssJs.style([|CssJs.media({js|screen, print and (color) |js}, [||])|])
+    ],
+  ),
+  (
+    "(min-height: 680px), screen and (orientation: portrait)",
+    [%expr
+      [%cx "@media (min-height: 680px), screen and (orientation: portrait) {}"]
+    ],
+    [%expr
+      CssJs.style([|
+        CssJs.media(
+          {js|(min-height: 680px), screen and (orientation: portrait) |js},
+          [||],
+        ),
+      |])
+    ],
+  ),
+  // (
+  //   "(not (color)) or (hover)",
+  //   [%expr [%cx "@media (not (color)) or (hover) {}"]],
+  //   [%expr
+  //     CssJs.style([|
+  //       CssJs.media({js|(not (color)) or (hover)|js}, [||]),
+  //     |])
+  //   ],
+  // ),
+  // (
+  //   "screen and (min-width: 33px) or (max-height: 15rem)",
+  //   [%expr
+  //     [%cx "@media screen and (min-width: 33px) or (max-height: 15rem) {}"]
+  //   ],
+  //   [%expr
+  //     CssJs.style([|
+  //       CssJs.media(
+  //         {js|screen and (min-width: 33px) or (max-height: 15rem)|js},
+  //         [||],
+  //       ),
+  //     |])
+  //   ],
+  // ),
+  (
+    "(max-height: $(wat)",
     [%expr [%cx "@media (max-height: $(wat)) {}"]],
     [%expr
       CssJs.style([|
-        CssJs.media({js|(max-height: |js} ++ wat ++ {js|)|js}, [||]),
+        CssJs.media({js|(max-height: |js} ++ wat ++ {js|) |js}, [||]),
       |])
     ],
   ),
-  /* Nested is not supported in code-gen
-     https://www.w3.org/TR/mediaqueries-5/#media-conditions */
-  /* (
-       "(not (color)) and (not (hover))",
-       [%expr [%cx "@media (not (color)) and (not (hover)) {}"]],
-       [%expr CssJs.style( [|CssJs.media( {js|((not (color)) and (not (hover))|js}, [||])|])]
-     ), */
```

Comment:
Can you keep this test? 

Is there an actual test passing with nesting?

Reply:
Yes, there is one and it passes!